WordPress/Classes/ViewRelated/Tools/SettingsListPickerViewController.swift

```diff
@@ -0,0 +1,160 @@
+import Foundation
+import WordPressShared
+
+
+
+/// SettingsListPicker will render a list of options, and will allow the user to select one from the list.
+///
+class SettingsListPickerViewController : UITableViewController
+{
+    /// Header Strings to be applied over the diferent sections
+    ///
+    var headers : [String]?
+
+    /// Footer Strings to be applied over the diferent sections
+    ///
+    var footers : [String]?
+    
+    /// Titles to be rendered
+    ///
+    var titles : [[String]]?
+
+    /// Row Subtitles. Should contain the exact same number as titles
+    ///
+    var subtitles : [[String]]?
```

Comment:
We're validating the arrays [in this spot](https://github.com/wordpress-mobile/WordPress-iOS/blob/issue/3558-language-interface-layer/WordPress/Classes/ViewRelated/Tools/SettingsListPickerViewController.swift#L48).

The developer should warrantee that they won't mismatch, so we're asserting on that condition. The goal is to crash as soon as possible, so that if you're implementing a new feature (and reusing this component)... and your input is incorrect, you'll notice in the first pass.

Does that work for you?